### PR TITLE
[OSE-301] Implement Create, Get, List, Delete endpoints for issuance templates

### PR DESCRIPTION
**Pull request by andresuribe87**
# Overview
This PR implements Create, Get, List, Delete endpoints for issuance templates.

# Description
This is very similar to the existing CRUD endpoints. The only main difference is that there are validation checks for some of the fields including the `schema`, `ID` and `expiry` of all the `IssuanceTemplate.Credentials` objects. 

# How Has This Been Tested?
Added tests for each of the endpoints. 

# Checklist

Before submitting this PR, please make sure:

- [x] I have read the CONTRIBUTING document.
- [x] My code is consistent with the rest of the project 
- [x] I have tagged the relevant reviewers and/or interested parties
- [x] I have updated the READMEs and other documentation of affected packages

## References
SIP5

### CODE REVIEW

File: pkg/server/router/issuance.go [hunk 1]
```diff
@@ -28,17 +31,35 @@ func NewIssuanceRouter(svc framework.Service) (*IssuanceRouter, error) {
...
+	return framework.Respond(ctx, w, issuanceTemplate.IssuanceTemplate, http.StatusOK)
 }
 
 type CreateIssuanceTemplateRequest struct {
 	issuing.IssuanceTemplate
 }
 
+func (r CreateIssuanceTemplateRequest) ToServiceRequest() *issuing.CreateIssuanceTemplateRequest {
+	return &issuing.CreateIssuanceTemplateRequest{
+		ID:               uuid.NewString(),
```
Comment (nitro-neal) [inline pkg/server/router/issuance.go:58]: In our other objects (like schema) we don't have the UUID in the request and we create it on more layer down in the CreateSchema function which is closer to DB.

Shouldn't matter either way, but eventally we should be consistent

File: pkg/server/router/issuance.go [hunk 2]
```diff
@@ -75,14 +120,22 @@ type ListIssuanceTemplatesResponse struct {
...
 // @Failure      500      {string}  string  "Internal server error"
 // @Router       /v1/manifests [get]
-func (ir IssuanceRouter) ListIssuanceTemplates(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
-	return nil
+func (ir IssuanceRouter) ListIssuanceTemplates(ctx context.Context, w http.ResponseWriter, _ *http.Request) error {
+	gotManifests, err := ir.service.ListIssuanceTemplates(&issuing.ListIssuanceTemplatesRequest{})
+
+	if err != nil {
+		return framework.NewRequestError(
+			util.LoggingErrorMsg(err, "could not get manifests"), http.StatusBadRequest)
```
Comment (nitro-neal) [inline pkg/server/router/issuance.go:136]: "could not get templates"

File: pkg/server/router/issuance.go [hunk 3]
```diff
@@ -65,8 +99,19 @@ func (ir IssuanceRouter) CreateIssuanceTemplate(ctx context.Context, w http.Resp
 // @Failure      400  {string}  string  "Bad request"
 // @Failure      500  {string}  string  "Internal server error"
 // @Router       /v1/issuancetemplates/{id} [delete]
-func (ir IssuanceRouter) DeleteIssuanceTemplate(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
-	return nil
+func (ir IssuanceRouter) DeleteIssuanceTemplate(ctx context.Context, w http.ResponseWriter, _ *http.Request) error {
+	id := framework.GetParam(ctx, IDParam)
+	if id == nil {
+		return framework.NewRequestError(
+			util.LoggingNewError("cannot delete a presentation without an ID parameter"), http.StatusBadRequest)
```
Comment (nitro-neal) [inline pkg/server/router/issuance.go:106]: template instead of presentation

File: pkg/server/router/issuance.go [hunk 4]
```diff
@@ -65,8 +99,19 @@ func (ir IssuanceRouter) CreateIssuanceTemplate(ctx context.Context, w http.Resp
...
+	}
+
+	if err := ir.service.DeleteIssuanceTemplate(&issuing.DeleteIssuanceTemplateRequest{ID: *id}); err != nil {
+		return framework.NewRequestError(
+			util.LoggingErrorMsgf(err, "could not delete presentation with id: %s", *id), http.StatusInternalServerError)
```
Comment (nitro-neal) [inline pkg/server/router/issuance.go:111]: template instead of presentation

**Review by nitro-neal — APPROVED** [review]
Looks good, some minor comments

File: pkg/server/server_issuance_test.go [hunk 5]
```diff
@@ -0,0 +1,472 @@
...
+			{
+				name: "returns a template with ID when schema is empty",
+				request: router.CreateIssuanceTemplateRequest{
+					IssuanceTemplate: issuing.IssuanceTemplate{
+						CredentialManifest: manifest.Manifest.ID,
+						Issuer:             issuerResp.DID.ID,
+						Credentials: []issuing.CredentialTemplate{
+							{
+								ID:     "output_descriptor_1",
+								Schema: "",
```
Comment (decentralgabe) [inline pkg/server/server_issuance_test.go:69]: remove

File: pkg/server/server_issuance_test.go [hunk 6]
```diff
@@ -0,0 +1,472 @@
...
+		}{
+			{
+				name: "when missing output_descriptor_id",
+				request: router.CreateIssuanceTemplateRequest{
+					IssuanceTemplate: issuing.IssuanceTemplate{
+						CredentialManifest: manifest.Manifest.ID,
+						Issuer:             issuerResp.DID.ID,
+						Credentials: []issuing.CredentialTemplate{
+							{
+								ID:     "",
```
Comment (decentralgabe) [inline pkg/server/server_issuance_test.go:118]: set or remove

File: pkg/service/issuing/model.go [hunk 7]
```diff
@@ -55,30 +52,33 @@ type CredentialTemplate struct {
...
-	// ID to be used when creating the issuance template. Must be unique.
-	// Required.
-	ID string `json:"id" validate:"required"`
-
 	// The template to create.
 	IssuanceTemplate IssuanceTemplate `json:"issuanceTemplate"`
 }
 
+func (r CreateIssuanceTemplateRequest) IsValid() bool {
+	return util.IsValidStruct(r) == nil
```
Comment (decentralgabe) [inline pkg/service/issuing/model.go:79]: why use this here and `Validate()` below - should they be consistent?

**Pull request merged**